x-pack/plugins/alerting/server/saved_objects/migrations/index.test.ts

```diff
@@ -2606,6 +2606,28 @@ describe('successful migrations', () => {
         expect(migratedAlert870.references).toEqual([]);
       });
     });
+
+    test('adds uuid to rule actions', () => {
+      const migration870 = getMigrations(encryptedSavedObjectsSetup, {}, isPreconfigured)['8.7.0'];
+      const rule = getMockData(
+        {
+          params: { foo: true },
+          alertTypeId: '.not-es-query',
+        },
+        true
+      );
+      const migratedAlert870 = migration870(rule, migrationContext);
+
+      expect(migratedAlert870.attributes.actions).toEqual([
+        {
+          group: 'default',
+          actionRef: '1',
+          actionTypeId: '1',
+          params: { foo: true },
+          uuid: expect.any(String),
```

Comment:
Since it's an optional field, a user can send any string.
Note: initially Create API was not accepting uuid, rather it was generation one. But we changed it to have a similar schema as the Update API.

Reply:
we can mock uuidv4 and expect its result but since it's used in some other tests here and  there isn't any extra benefit, i tested like this.